superset-frontend/src/SqlLab/components/ResultSet.tsx

```diff
@@ -404,7 +404,7 @@ export default class ResultSet extends React.PureComponent<
       query.errorMessage &&
       query.errorMessage.indexOf('session timed out') > 0
     ) {
-      this.props.actions.runQuery(query, true);
+      this.props.actions.reRunQuery(query);
```

Comment:
Oh, this is why the query was re-run after 1 minute with the same ID! Good catch!

Reply:
I'm hoping that was it! :) I haven't been able to prove it yet.